source/opt/fold.cpp

```diff
@@ -199,28 +193,21 @@ bool FoldInstructionInternal(ir::Instruction* inst) {
   std::vector<const analysis::Constant*> constants =
       const_manager->GetOperandConstants(inst);
 
-  static FoldingRules* rules = new FoldingRules();
-  for (FoldingRule rule : rules->GetRulesForOpcode(opcode)) {
+  for (FoldingRule rule : GetFoldingRules().GetRulesForOpcode(opcode)) {
```

Comment:
Drive by: Would it be useful if this were a FoldingRule reference?

Reply:
A folding rule is a function object.  Not something large.  I don't think it will make a difference.